src/contracts/middleware/IndexRegistry.sol

```diff
@@ -211,12 +215,9 @@ contract IndexRegistry is IIndexRegistry {
         if (totalOperatorsHistoryLength == 0) {
             return 0;
         }
-                
-        // if `blockNumber` is in future, return current number of operators
-        if (blockNumber > uint32(block.number)) {
-            return _totalOperatorsHistory[quorumNumber][totalOperatorsHistoryLength - 1].index;
-        // else if `blockNumber` is from before the `quorumNumber` existed, return `0`
-        } else if (blockNumber < _totalOperatorsHistory[quorumNumber][0].fromBlockNumber) {
```

Comment:
does removing this 'if' block make the comment about the function's behavior inaccurate? specifically `returns the current number of total operators if the @param blockNumber is in the future.`

Reply:
i think looping backwards through the stake updates should achieve this behavior. am i mistaken?

Reply:
looks like you switched the handedness of the check in the loop? will comment there

Reply:
> looks like you switched the handedness of the check in the loop? will comment there

actually nevermind, I see that it was incorrect before and simply fixed now. resolving this thread.